python: make Assembly constructible and mutable

keitaroyam opened the pull request:
Maybe there is a better way?

wojdyr commented:
It looks good apart from:

    PYBIND11_MAKE_OPAQUE(std::vector<std::string>)

The problem with it is that it'd need to be added in every file that has vector<string> in the interface. Can we do without it?

It's also worth to test if pydoc documentation is fine, which depends on the order of added bindings. At this moment I test it with:

    $PYTHON -m pydoc gemmi | grep ::

if the order is wrong, some C++ symbols end up in pydoc. (in the future it will be tested automatically)

keitaroyam commented:
Thank you. I removed it (and StrList) and changed (sub)chains to def_readwrite(). Is this ok?

I also found that the updated assembly would not be written to a pdb file if raw_remarks was not empty. For mmcif it is ok.
https://github.com/project-gemmi/gemmi/blob/5ad0146fbbd080b4b58b02aabafa86f97b834679/include/gemmi/to_pdb.hpp#L412

wojdyr commented:
Thanks!
I don't have a good solution for `raw_remarks`. Currently, either all remarks are written as the were in the input file, or only those remarks are written that can be parsed, stored and formatted back as a string. It'd be hard to mix the two approaches.
So in your case you'd need to add `st.raw_remarks = []`.
I'll add a note about it in documentation.
